include/jinja2cpp/string_helpers.h

```diff
@@ -0,0 +1,204 @@
+#ifndef JINJA2_STRING_HELPERS_H
+#define JINJA2_STRING_HELPERS_H
+
+#include <nonstd/string_view.hpp>
+#include "value.h"
+
+#include <string>
+#include <type_traits>
+#include <cwchar>
+
+namespace jinja2
+{
+namespace detail
+{
+    template<typename Src, typename Dst>
+    struct StringConverter;
+
+    template<typename Src>
+    struct StringConverter<Src, Src>
+    {
+        static Src DoConvert(const nonstd::basic_string_view<typename Src::value_type>& from)
+        {
+            return Src(from.begin(), from.end());
+        }
+    };
+
+    template<>
+    struct StringConverter<std::wstring, std::string>
+    {
+        static std::string DoConvert(const nonstd::wstring_view& from)
+        {
+            std::mbstate_t state = std::mbstate_t();
+            auto srcPtr = from.data();
+            std::size_t srcSize = from.size();
+            std::size_t destBytes = 0;
+
+#ifndef _MSC_VER
+            destBytes = std::wcsrtombs(nullptr, &srcPtr, srcSize, &state);
+            if (destBytes == static_cast<std::size_t>(-1))
+                return std::string();
```

Comment:
Isn't it better to throw exception here? in such a way we silently swallow error

Reply:
And what to do with this exception? :) I'll think about it, actually.